public/app/features/dashboard/panel_editor/QueryEditorRows.tsx

```diff
@@ -0,0 +1,95 @@
+// Libraries
+import React, { PureComponent } from 'react';
+import _ from 'lodash';
+
+// Types
+import { PanelModel } from '../state/PanelModel';
+import { DataQuery, PanelData, DataSourceSelectItem } from '@grafana/ui';
+import { DashboardModel } from '../state/DashboardModel';
+import { QueryEditorRow } from './QueryEditorRow';
+import { addQuery } from 'app/core/utils/query';
+
+interface Props {
+  // The query configuration
+  queries: DataQuery[];
+  datasource: DataSourceSelectItem;
+
+  // Query editing
+  onChangeQueries: (queries: DataQuery[]) => void;
+  onScrollBottom: () => void;
+
+  // Dashboard Configs
+  panel: PanelModel;
+  dashboard: DashboardModel;
+
+  // Query Response Data
+  data: PanelData;
+}
+
+export class QueryEditorRows extends PureComponent<Props> {
+  onAddQuery = (query?: Partial<DataQuery>) => {
+    const { queries, onChangeQueries } = this.props;
+    onChangeQueries(addQuery(queries, query));
+    this.props.onScrollBottom();
+  };
+
+  onRemoveQuery = (query: DataQuery) => {
+    const { queries, onChangeQueries, panel } = this.props;
+    const removed = queries.filter(q => {
+      return q !== query;
+    });
+    onChangeQueries(removed);
+    console.log('REMOVE (QueryEditorRows)', removed, queries);
+    panel.refresh();
+  };
+
+  onMoveQuery = (query: DataQuery, direction: number) => {
+    const { queries, onChangeQueries, panel } = this.props;
+
+    const index = _.indexOf(queries, query);
+    // @ts-ignore
+    _.move(queries, index, index + direction);
```

Comment:
Is there any way to avoid lodash? If not try to direct import functions instead of the whole `_`